private/bufpkg/bufcheck/buflint/buflint_test.go

```diff
@@ -614,6 +614,7 @@ func TestRunProtovalidateRules(t *testing.T) {
 		bufanalysistesting.NewFileAnnotation(t, "map.proto", 44, 5, 44, 47, "PROTOVALIDATE"),
 		bufanalysistesting.NewFileAnnotation(t, "map.proto", 46, 5, 46, 47, "PROTOVALIDATE"),
 		bufanalysistesting.NewFileAnnotation(t, "map.proto", 50, 5, 50, 57, "PROTOVALIDATE"),
+		bufanalysistesting.NewFileAnnotation(t, "map.proto", 53, 5, 53, 50, "PROTOVALIDATE"),
```

Comment:
Should this produce two lint errors l, one for each of the two mismatched annotations?

Reply:
The two lines belong to the same rule, `(buf.validate.field).map`. 

When we pass the path `(buf.validate.field).map` to `OptionExtensionLocation`, the location we get back is just the first line.

I think it's due to the way `sourceCodeLocation`s are generated by the compiler -- there isn't a location that spans both lines exactly.

Reply:
We are also not able to tell whether the option is specified in two lines or in the aggregate form.

On a high level, the linter cannot tell

```
  int32 not_a_map = 14 [
    (buf.validate.field).map.values.int32.lt = 10,
    (buf.validate.field).map.values.int32.gt = 1
  ];
```

from

```
  int32 not_a_map = 14 [(buf.validate.field).map.values.int32 = {lt: 10, gt: 1}];
```